includes/HttpFunctions.php

```diff
@@ -430,6 +428,12 @@ public function getHeaderList() {
 				);
 		}
 
+		// SUS-5499: Use internal host name for MW->MW requests when running on Kubernetes
+		global $wgKubernetesNamespace;
+		if ( !empty( $wgKubernetesNamespace ) && Http::isLocalURL( $this->url ) ) {
+			$list[] = sprintf( 'X-Original-Host: %s', $this->parsedUrl['host'] );
```

Comment:
It smells a bit to have an `if` like that but I don't have a better idea where to put it.

Reply:
You know me, I like smelly things ;)